src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -74,10 +74,13 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
                 _textView.Options.GlobalOptions.SetOptionValue(NonBlockingCompletionEditorOption, true);
             }
 
-            if (!triggerLocation.Snapshot.TextBuffer.Properties.ContainsProperty(PotentialCommitCharacters))
-            {
-                triggerLocation.Snapshot.TextBuffer.Properties.AddProperty(PotentialCommitCharacters, service.GetRules().DefaultCommitCharacters);
-            }
+            var commitCharacters = service.GetRules().DefaultCommitCharacters;
+
+            // In case of calls with multiple completion services for the same view (e.g. TypeScript and C#), we should merge potential commit characters together.
+            _textView.Properties[PotentialCommitCharacters] =
+                _textView.Properties.TryGetProperty(PotentialCommitCharacters, out ImmutableHashSet<char> savedCommitCharacters)
+                ? savedCommitCharacters.Union(commitCharacters)
+                : commitCharacters.ToImmutableHashSet();
```

Comment:
when things are indented at the same level, it makes it seem like it's at the same nesting level.  that's not the case here.  The `?` and `:` refer to the `_textView.Properties.TryGetProperty` expression, not the `             _textView.Properties[PotentialCommitCharacters]` expression.